api/controllers/bookmark.js

```diff
@@ -83,59 +94,56 @@ router.get("/list/:userId", function(req, res, next) {
  */
 
 router.post("/add", function addBookmark(req, res, next) {
-  try {
-    if (!req.body.bookmarkType) {
-      log.error("Required parameter (bookmarkType) wasn't specified");
-      res.status(400).json({
-        message: "Required parameter (bookmarkType) wasn't specified"
+  if (!req.body.bookmarkType) {
```

Comment:
We should validate and escape bookmarkType and thingId (userId too, but it is relatively safe after passing through our authentication process.